### ISSUE TITLE
rdc.h doesn't work on gcc 4 or gcc 8 with rhel7

### ISSUE THREAD

**Issue by baallan** [RadeonOpenCompute/ROCm#1641]
The definition of a const MAX_TEST_CASES in rdc.h causes gcc to fail with the error below.
One solution is illustrated by:
```
sed -i -e 's/^const uint32_t MAX_TEST_CASES =  RDC_DIAG_TEST_LAST - RDC_DIAG_TEST_FIRST + 1;/#define MAX_TEST_CASES (RDC_DIAG_TEST_LAST - RDC_DIAG_TEST_FIRST + 1)/' $(RDC_INC)/rdc/rdc.h
```

the error is:
```
/opt/rocm-4.5.0/rdc/include/rdc/rdc.h:479:28: error: variably modified 'diag_info' at file scope
     rdc_diag_test_result_t diag_info[MAX_TEST_CASES];
                            ^
```

Whether or not this is an excessive error detection by gcc I cannot say, but on my production hpc clusters gcc is what we compile with.

**Comment by ROCmSupport** [comment]
Hi @baallan 
Thanks for reaching out.
Let me try to gather some information on this.

**Comment by ROCmSupport** [comment]
Hi @baallan 
I am not able to reproduce locally with ROCm 5.1. Can you check and comeback asap. Thank you.

**Comment by baallan** [comment]
The latest rocm on any of my systems is 4.5.0.
If you have a pointer to an rdc.h source repo corresponding to rocm 5.1, I can try downloading and compiling.

Ben

From: ROCmSupport ***@***.***>
Sent: Sunday, May 8, 2022 10:52 PM
To: RadeonOpenCompute/ROCm ***@***.***>
Cc: Allan, Benjamin ***@***.***>; Mention ***@***.***>
Subject: [EXTERNAL] Re: [RadeonOpenCompute/ROCm] rdc.h doesn't work on gcc 4 or gcc 8 with rhel7 (Issue #1641)


Hi @baallan<https://gcc02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fbaallan&data=05%7C01%7Cbaallan%40sandia.gov%7C43bc6db9f5d24d2f82ce08da31779793%7C7ccb5a20a303498cb0c129007381b574%7C1%7C0%7C637876686993337439%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=DndMnrd%2FItzB4KrRwp4suDAs74WrbjYfMSuO%2BxxGBV8%3D&reserved=0>
I am not able to reproduce locally with ROCm 5.1. Can you check and comeback asap. Thank you.

-

**Comment by morrone** [comment]
This is still a problem in rocm-5.2.1.

Here is a stackoverflow entry describing why this doesn't work with gcc in more detail.

The long and short of it is that "const" just means "read only". It does not mean determined-at-compile-time. It is still a variable. The use of MAX_TEST_CASES needs to be a a fixed value at compile time, not merely a read-only variable (a const).

The problem goes away if MAX_TEST_CASES is a #define, as @baallan pointed out.

**Comment by dmitrii-galantsev** [comment]
fixed as of 5.5 https://github.com/RadeonOpenCompute/rdc/commit/73d8b35610564cd6cac03ab6cef3882f5ac0215f
